daphne/src/lib.rs

```diff
@@ -304,18 +304,80 @@ pub struct DapGlobalConfig {
     pub supported_hpke_kems: Vec<HpkeKemId>,
 }
 
+/// DAP Query configuration.
+//
+// TODO(cjpatton) Once we implement maximum batch lifetime, put the parameter here.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum DapQueryConfig {
+    /// The "time-interval" query type. Each report in the batch must fall into the time interval
+    /// specified by the query.
+    TimeInterval { min_batch_size: u64 },
+
+    /// The "fixed-size" query type. The Leader partitions the reports into arbitary batches of
+    /// roughly the same size.
+    FixedSize {
+        min_batch_size: u64,
+        max_batch_size: u64,
+    },
+}
+
+impl DapQueryConfig {
+    /// Check if the batch is ready to aggregate based on the report count. Returns an error if the
+    /// batch size is too large.
```

Comment:
Isn't the error returned if the _report count_ is too large (`report_count > *max_batch_size`)?

Reply:
Yup! Done.